server/src/test/java/org/elasticsearch/search/internal/ContextIndexSearcherTests.java

```diff
@@ -376,6 +380,38 @@ public void onRemoval(ShardId shardId, Accountable accountable) {
         IOUtils.close(reader, w, dir);
     }
 
+    public void testComputeSlices() throws IOException {
+        Directory dir = newDirectory();
+        RandomIndexWriter w = new RandomIndexWriter(random(), dir);
+        int numDocs = randomIntBetween(1000, 25000);
```

Comment:
lets also rarely do something like this: `rarely() ? randomIntBetween(0, 1000) : randomIntBetween(1000, 25000);`

Reply:
good point, I have actually check manually the behaviour for 0 documents and as expected it returns an empty array.